lm_eval/api/task.py

```diff
@@ -1087,16 +1098,22 @@ def fewshot_context(
             else:
                 labeled_examples += self.sampler.get_context(doc, num_fewshot)
 
+        ## [Prompt] ##
         example = self.doc_to_text(doc)
         if apply_chat_template:
             if self.multiple_input:
+                # TODO<baber>: How to handle gen_prefix for multiple inputs?
                 return chat_template(labeled_examples)
             if isinstance(example, str):
                 self.append_target_question(
-                    labeled_examples, example, fewshot_as_multiturn
+                    labeled_examples,
+                    example,
+                    fewshot_as_multiturn,
+                    gen_prefix,
                 )
             # for loglikelihood create a list of questions with appended choices
             elif isinstance(example, list):
```

Comment:
Haven't been able to identify when non-chat_template `example` is a list

Reply:
it would be better to split fewshot_context func into two: one to make a list of samples, another to make it a valid string
right now i am working on smart truncation that allows users to choose what to truncate and I encountered the same problem